src/runtime/server/middleware/session/index.ts

```diff
@@ -8,18 +8,26 @@ import { SessionExpired } from './exceptions'
 import { useRuntimeConfig } from '#imports'
 
 const SESSION_COOKIE_NAME = 'sessionId'
-const safeSetCookie = (event: H3Event, name: string, value: string) => setCookie(event, name, value, {
-  // Max age of cookie in seconds
-  maxAge: useRuntimeConfig().session.session.expiryInSeconds,
-  // Only send cookie via HTTPs to mitigate man-in-the-middle attacks
-  secure: true,
-  // Only send cookie via HTTP requests, do not allow access of cookie from JS to mitigate XSS attacks
-  httpOnly: true,
-  // Do not send cookies on many cross-site requests to mitigates CSRF and cross-site attacks, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#lax
-  sameSite: useRuntimeConfig().session.session.cookieSameSite as SameSiteOptions,
-  // Set cookie for subdomain
-  domain: useRuntimeConfig().session.session.domain
-})
+const safeSetCookie = (event: H3Event, name: string, value: string, date: Date) => {
+  const sessionConfig = useRuntimeConfig().session.session
+  const expirationDate = sessionConfig.expiryInSeconds ? date : undefined
+  if (expirationDate) {
+    expirationDate.setSeconds(expirationDate.getSeconds() + sessionConfig.expiryInSeconds)
+  }
```

Comment:
It's never a good idea to modify input arguments of a function (in this case `date`), especially when it's an object (so it will produce side effects outside of the function). 

Also, nit: do not rename one thing into another (`date` -> `expirationDate` since it it's not yet `expirationDate` when it got renamed and it harder to read and can lead to confusion)

Instead, create a new date:
```suggestion
  const expirationDate = sessionConfig.expiryInSeconds
      ? new Date(createdAt.getTime() + sessionConfig.expiryInSeconds * 1000)
      : undefined
```